### PR TITLE
Omni-Node renamings

### PR DESCRIPTION
**Pull request by serban300**
- moved the omni-node lib from
  `cumulus/polkadot-parachain/polkadot-parachain-lib` to
  `cumulus/polkadot-omni-node/lib`
- renamed `polkadot-parachain-lib` to `polkadot-omni-node-lib`
- added `polkadot-omni-node` binary

Related to https://github.com/paritytech/polkadot-sdk/issues/5566

### CODE REVIEW

**Comment by skunert** [comment]
I am not entirely sure if we gain something from this naming. To us its maybe obvious. But looking from the outside, `polkadot-parachain` or even just `parachain-node` sounds more intuitive and less surprising to me. In the end this is exactly what it is, just a `parachain-node`.

**Comment by serban300** [comment]
> I am not entirely sure if we gain something from this naming. To us its maybe obvious. But looking from the outside, `polkadot-parachain` or even just `parachain-node` sounds more intuitive and less surprising to me. In the end this is exactly what it is, just a `parachain-node`.

@kianenigma what do you think ?

**Comment by kianenigma** [comment]
@xlc as you are the only user of the library I'm afraid this will be a breaking change for you in https://github.com/AcalaNetwork/acala-node. But we promise this is the last one 🤝

**Comment by kianenigma** [comment]
(have not read the code yet)

I am not surprised to see pushback on the renaming 🙈 

I am in favor of the renaming. Here's my reasons: 

1. It amplifies the code-word of the project "omni-_something_". It is a guess, but I think it is more comprehensible to repeat the omni-prefix in all such tools.
2. Calling it `parachain-node` and/or `polkadot-parachain` prevents us from ever even _imagining_ adding more abilities related to being a solo-chain to this node, something that I [guess even parachain teams reasonably want](https://github.com/orgs/paritytech/projects/157/views/1?pane=issue&itemId=82036322) esp. to run a testnet. 
3. It changes nothing about polkadot-parachain: It remains as is, and it is indeed "Just a polkadot parachain node" as @skunert puts it. 

Perhaps the confusion is partly because my original issue was lacking on point (remove any chainspec from the generic polkadot-omni-node). The outcome of this PR should be: 

0. rename the library crate to `polkadot-omni-node-lib`. This is hidden from all users atm, because no one is using this crate yet (unless if you are @xlc and `acala-node`). 
1. `polkadot-parachain` remains 100% as-is, and it is an instantiation of `polkadot-omni-node-lib` with all polkadot-related chainspecs hardcoded. This makes perfect sense to me in terms of naming: A parachain node, built from the generic `polkadot-omni-node-lib`, that is specialized for running Polkadot system chains, ergo `polakdot-parachain`. 
2. `polkadot-omni-node` is a new binary, which is a raw instantiation of `polkadot-omni-node-lib` without any chainspecs or specialization. 

So when looked at this way, we are not really touching `polkadot-parachain`. It remains as-is, and we are adding a new binary that is the more generic version of it.

File: Cargo.toml [hunk 1]
```diff
@@ -135,8 +135,8 @@ members = [
 	"cumulus/parachains/runtimes/test-utils",
 	"cumulus/parachains/runtimes/testing/penpal",
 	"cumulus/parachains/runtimes/testing/rococo-parachain",
-	"cumulus/polkadot-parachain",
-	"cumulus/polkadot-parachain/polkadot-parachain-lib",
+	"cumulus/polkadot-omni-node/polkadot-omni-node-lib",
+	"cumulus/polkadot-omni-node/polkadot-parachain",
```
Comment (kianenigma) [inline Cargo.toml:139]: So just looking at this file, i can confirm that this is not what I intended (as noted, my original issue was not super clear). High level, expect to see something like: 

`cumulus/polkadot-parachain` should stay. This is also crucial for our internal devops to not put a bounty on my head. 
`cumulus/polkadot-omni-node` should be a new crate. 
`cumulus/polkadot-parachain-lib` ->  `cumulus/polkadot-omni-node/lib`

**Comment by skunert** [comment]
> (have not read the code yet)
> 
> I am not surprised to see pushback on the renaming 🙈
> 
> I am in favor of the renaming. Here's my reasons:
> 
>     1. It amplifies the code-word of the project "omni-_something_". It is a guess, but I think it is more comprehensible to repeat the omni-prefix in all such tools.
> 
>     2. Calling it `parachain-node` and/or `polkadot-parachain` prevents us from ever even _imagining_ adding more abilities related to being a solo-chain to this node, something that I [guess even parachain teams reasonably want](https://github.com/orgs/paritytech/projects/157/views/1?pane=issue&itemId=82036322) esp. to run a testnet.
> 
>     3. It changes nothing about polkadot-parachain: It remains as is, and it is indeed "Just a polkadot parachain node" as @skunert puts it.
> 
> 
> Perhaps the confusion is partly because my original issue was lacking on point (remove any chainspec from the generic polkadot-omni-node). The outcome of this PR should be:
> 
>     0. rename the library crate to `polkadot-omni-node-lib`. This is hidden from all users atm, because no one is using this crate yet (unless if you are @xlc and `acala-node`).
> 
>     1. `polkadot-parachain` remains 100% as-is, and it is an instantiation of `polkadot-omni-node-lib` with all polkadot-related chainspecs hardcoded. This makes perfect sense to me in terms of naming: A parachain node, built from the generic `polkadot-omni-node-lib`, that is specialized for running Polkadot system chains, ergo `polakdot-parachain`.
> 
>     2. `polkadot-omni-node` is a new binary, which is a raw instantiation of `polkadot-omni-node-lib` without any chainspecs or specialization.
> 
> 
> So when looked at this way, we are not really touching `polkadot-parachain`. It remains as-is, and we are adding a new binary that is the more generic version of it.

These are good points, I also forgot about `frame-omni-bencher` which goes nicely with this.

File: Cargo.toml (hunk 1 above)
Comment (serban300) [reply Cargo.toml:139]: Sorry, I misunderstood the requirement a bit. Will fix it.

**Comment by kianenigma** [comment]
> These are good points, I also forgot about frame-omni-bencher which goes nicely with this.

In my dreams, the binary list that one would see in our release pages would be:

1. `polkadot-parachain`
2. `polkadot-omni-node`
3. `polkadot-omni-bencher`
4. `polkadot-omni-chain-spec-builder`

Not to late to rename them? idk. But given this item in the roadmap, I won't push for this. Goal is to allow users to to all of this in just `polkadot-omni-node`.

@serban300 another small action item that comes to mind here is to ensure that as a consequence of this PR, `polkadot-omni-node` is also released. This requires a few GH/GL action files to be edited, https://github.com/paritytech/polkadot-sdk/pull/5387 is probably a good example. CC @EgorPopelyaev

**Comment by serban300** [comment]
> So just looking at this file, i can confirm that this is not what I intended (as noted, my original issue was not super clear). High level, expect to see something like:
> 
> `cumulus/polkadot-parachain` should stay. This is also crucial for our internal devops to not put a bounty on my head. `cumulus/polkadot-omni-node` should be a new crate. `cumulus/polkadot-parachain-lib` -> `cumulus/polkadot-omni-node/lib`

@kianenigma @skunert I addressed this comment. I had to force push because there were a lot of conflicts. Can you PTAL ? his PR only contains the renamings and also added the `polkadot-omni-node` binary. I will implement the part about publishing the `polkadot-omni-node` binary in a different PR. Would like to merge this one as soon as possible to avoid merge conflicts since it moves a lot of files.

File: prdoc/pr_5915.prdoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Omni-Node renamings
+
+doc:
+  - audience: Node Dev
+    description: |
+      This PR renames the `polkadot-parachain-lib` crate to `polkadot-omni-node-lib`.
```
Comment (skunert) [inline prdoc/pr_5915.prdoc:9]: We could mention the introduction of `polkadot-omni-node` binary.

**Review by skunert — APPROVED** [review]
Nice!

File: prdoc/pr_5915.prdoc (hunk 2 above)
Comment (serban300) [reply prdoc/pr_5915.prdoc]: Good point ! Done.

File: cumulus/polkadot-omni-node/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "polkadot-omni-node"
+version = "0.1.0"
+authors.workspace = true
+edition.workspace = true
+build = "build.rs"
+description = "Runs a polkadot parachain node"
```
Comment (kianenigma) [inline cumulus/polkadot-omni-node/Cargo.toml:7]: Better description please.

File: cumulus/polkadot-omni-node/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Basic polkadot omni-node.
+//!
+//! It can be used to start a parachain node from a provided chain spec file.
+//! It is only compatible with runtimes that use block number `u32` and `Aura` consensus.
```
Comment (kianenigma) [inline cumulus/polkadot-omni-node/src/main.rs:20]: Example usage would be great.

**Review by kianenigma — APPROVED** [review]

File: cumulus/polkadot-omni-node/Cargo.toml (hunk 3 above)
Comment (serban300) [reply cumulus/polkadot-omni-node/Cargo.toml]: Done

File: cumulus/polkadot-omni-node/src/main.rs (hunk 4 above)
Comment (serban300) [reply cumulus/polkadot-omni-node/src/main.rs:20]: Added

**Pull request merged**